src/makeInfinityAdder.js

```diff
@@ -7,6 +7,21 @@
 
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  return function adder(x) {
+    if (x === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += x;
+
+    return adder;
+  };
```

Comment:
better use anonymous function
and no need to check for `undefined` - just provide an initial value
```suggestion
  return adder = (x = 0) {
    ...
  };
```

Reply:
i forgot an actual arrow for an arrow function
did you use it?
`return adder = (x = 0) => {`
weird that you've got lots or errors.

Yet we could keep your code as is.

Reply:
Yes I added the => but didn't work! 

Thank you :)